src/commons/navbar-searchbar/navbar/navbar.js

```diff
@@ -5,37 +5,20 @@ import { EventAggregator } from 'aurelia-event-aggregator';
 import { Router } from 'aurelia-router';
 import { safeGet } from '../../util/utility';
 import './navbar.scss';
-import { WorkspaceService } from '../../workspaces-menu/workspace-service';
-import * as toastr from 'toastr';
 import { AppService } from '../../services/app-service';
 import { NavigationService } from '../../services/navigation-service';
 import { faTh } from '@fortawesome/free-solid-svg-icons';
 
-@inject(
-  Router,
-  EventAggregator,
-  WorkspaceService,
-  Store,
-  AppService,
-  NavigationService
-)
+@inject(Router, EventAggregator, Store, AppService, NavigationService)
 export class Navbar {
-  constructor(
-    router,
-    ea,
-    workspaceService,
-    store,
-    appService,
-    navigationService
-  ) {
+  constructor(router, ea, store, appService, navigationService) {
     this.store = store;
     this.store.registerAction('reset', reset);
     this.subscription = this.store.state.subscribe(
```

Comment:
Make sure that you did not forget to unsubscribe

Reply:
aren't we doing this in detached lifecycle hook? :)
```javascript
detached() {
    this.routerSubscription.unsubscribe();
    this.subscription.unsubscribe();
  }

```

Reply:
yep, it is - haven't noticed